### PR TITLE
DOC: warning on look-ahead bias with resampling

### LINKED ISSUE #11123: Business day resampling
Hi everyone, 

I just stumbled across this odd behaviour: 

For a daily time series, I would expect `resample('B',  how='last')` to be equivalent to selecting rows whose weekday is not Saturday or Sunday.  However, `resample` groups together Fridays,  Saturdays and Sundays and hence we get Sunday's value on Friday.

In the time series context, I guess most people would expect `resample('B', ...)` to bin Saturdays, Sundays and Mondays, so that **no looking into the future** occurs.

Am I missing something?

### PR DESCRIPTION
**Pull request by 0x0L**
- [x] closes #11123
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -1628,24 +1600,29 @@ labels.
...
 
-.. note::
+.. warning::
 
-    The default values for ``label`` and ``closed`` is 'left' for all
+    The default values for ``label`` and ``closed`` is '**left**' for all
     frequency offsets except for 'M', 'A', 'Q', 'BM', 'BA', 'BQ', and 'W'
     which all have a default of 'right'.
 
+    This might lead to unintended look-ahead bias as in the following example:
```
Comment (TomAugspurger) [inline doc/source/user_guide/timeseries.rst]: Is "look-ahead bias" a common term?

Comment (0x0L) [reply doc/source/user_guide/timeseries.rst]: It is in the context of timeseries forecasting

Comment (TomAugspurger) [reply doc/source/user_guide/timeseries.rst]: Dropping bias, and perhaps adding a small definition. Something like

    This might lead to unintended looking ahead, where the value for a later date (Sunday) is pulled back to
    a previous date (Friday).

File: doc/source/user_guide/timeseries.rst [hunk 2]
```diff
@@ -1796,6 +1776,34 @@ level of ``MultiIndex``, its name or location can be passed to the
 
    df.resample('M', level='d').sum()
 
+.. _timeseries.iterating-label:
```
Comment (jreback) [inline doc/source/user_guide/timeseries.rst]: why did you move this? (not averse, just want to know the reason)

Comment (0x0L) [reply doc/source/user_guide/timeseries.rst]: It looked out of place where it was. I put with the rest of the resampling related

**Comment by jreback** [comment]
thanks @0x0L
